src/dxvk/dxvk_swapchain_blitter.cpp

```diff
@@ -0,0 +1,372 @@
+    ctx->draw(3, 1, 0, 0);
+    ctx->setSpecConstant(VK_PIPELINE_BIND_POINT_GRAPHICS, 0, 0);
+  }
+
+  void DxvkSwapchainBlitter::resolve(
+          DxvkContext*        ctx,
+    const Rc<DxvkImageView>&  dstView,
+    const Rc<DxvkImageView>&  srcView) {
+    VkImageResolve resolve;
+    resolve.srcSubresource = { VK_IMAGE_ASPECT_COLOR_BIT, 0, 0, 1 };
+    resolve.srcOffset      = { 0, 0, 0 };
+    resolve.dstSubresource = { VK_IMAGE_ASPECT_COLOR_BIT, 0, 0, 1 };
+    resolve.dstOffset      = { 0, 0, 0 };
+    resolve.extent         = dstView->imageInfo().extent;
+    ctx->resolveImage(dstView->image(), srcView->image(), resolve, VK_FORMAT_UNDEFINED);
+  }
+
+
+  void DxvkSwapchainBlitter::updateGammaTexture(DxvkContext* ctx) {
+    uint32_t n = uint32_t(m_gammaRamp.size());
+
+    if (n) {
+      // Reuse existing image if possible
+      if (m_gammaImage == nullptr || m_gammaImage->info().extent.width != n) {
+        DxvkImageCreateInfo imgInfo;
+        imgInfo.type        = VK_IMAGE_TYPE_1D;
+        imgInfo.format      = VK_FORMAT_R16G16B16A16_UNORM;
+        imgInfo.flags       = 0;
+        imgInfo.sampleCount = VK_SAMPLE_COUNT_1_BIT;
+        imgInfo.extent      = { n, 1, 1 };
+        imgInfo.numLayers   = 1;
+        imgInfo.mipLevels   = 1;
+        imgInfo.usage       = VK_IMAGE_USAGE_TRANSFER_DST_BIT
+                            | VK_IMAGE_USAGE_SAMPLED_BIT;
+        imgInfo.stages      = VK_PIPELINE_STAGE_TRANSFER_BIT
+                            | VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT;
+        imgInfo.access      = VK_ACCESS_TRANSFER_WRITE_BIT
+                            | VK_ACCESS_SHADER_READ_BIT;
+        imgInfo.tiling      = VK_IMAGE_TILING_OPTIMAL;
+        imgInfo.layout      = VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL;
+        
+        m_gammaImage = m_device->createImage(
+          imgInfo, VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT);
+
+        DxvkImageViewCreateInfo viewInfo;
+        viewInfo.type       = VK_IMAGE_VIEW_TYPE_1D;
+        viewInfo.format     = VK_FORMAT_R16G16B16A16_UNORM;
+        viewInfo.usage      = VK_IMAGE_USAGE_SAMPLED_BIT;
+        viewInfo.aspect     = VK_IMAGE_ASPECT_COLOR_BIT;
+        viewInfo.minLevel   = 0;
+        viewInfo.numLevels  = 1;
+        viewInfo.minLayer   = 0;
+        viewInfo.numLayers  = 1;
+        
+        m_gammaView = m_device->createImageView(m_gammaImage, viewInfo);
+      }
+
+      ctx->updateImage(m_gammaImage,
+        VkImageSubresourceLayers { VK_IMAGE_ASPECT_COLOR_BIT, 0, 0, 1 },
+        VkOffset3D { 0, 0, 0 },
+        VkExtent3D { n, 1, 1 },
+        m_gammaRamp.data(),
+        sizeof(DxvkGammaCp) * n,
+        sizeof(DxvkGammaCp) * n);
+    } else {
+      m_gammaImage = nullptr;
+      m_gammaView  = nullptr;
+    }
+
+    m_gammaDirty = false;
+  }
+
+
+  void DxvkSwapchainBlitter::createSampler() {
+    DxvkSamplerCreateInfo samplerInfo;
+    samplerInfo.magFilter       = VK_FILTER_LINEAR;
+    samplerInfo.minFilter       = VK_FILTER_LINEAR;
```

Comment:
We used to use _NEAREST when the extents matched, do we still want to do that maybe?

I know we are using pixelCoord, but...

Reply:
We're using texelFetch in that.

Reply:
Got that after looking again, and realised that was only on the blit path LGTM then!